app/src/main/java/zapsolutions/zap/connection/establishConnectionToLnd/LndConnection.java

```diff
@@ -49,45 +54,26 @@ public static synchronized LndConnection getInstance() {
 
     private void readSavedConnectionInfo() {
 
-        App ctx = App.getAppContext();
+        // Load current wallet connection config
+        mConnectionConfig = WalletConfigsManager.getInstance().loadCurrentWalletConfig();
 
-        mPrefsRemote = Armadillo.create(ctx, PrefsUtil.PREFS_REMOTE)
-                .encryptionFingerprint(ctx)
-                .keyStretchingFunction(new PBKDF2KeyStretcher(RefConstants.NUM_HASH_ITERATIONS, null))
-                .password(ctx.inMemoryPin.toCharArray())
-                .contentKeyDigest(UtilFunctions.getZapsalt().getBytes())
-                .build();
+        // Generate Macaroon
+        mMacaroon = new MacaroonCallCredential(mConnectionConfig.getMacaroon());
```

Comment:
My app is crashing here as I don't have a walletconfig at startup, so `mConnectionConfig` is null.
I guess it is because we didn't update `CURRENT_SETTINGS_VER` yet, and you mentioned you will probably add some converting functionality for this.